tests/core/test_utils.py

```diff
@@ -11,3 +23,227 @@ def test_static_directory_provider():
+    assert logs[0].levelname == "DEBUG"
+
+
+async def test_error_handling_value_errors(caplog):
+    """Checks that NotConnected is aggregated correctly across Devices."""
+
+    caplog.set_level(10)
+
+    dummy_device_two_working_one_timeout_two_value_error = (
+        DummyDeviceTwoWorkingTwoTimeOutTwoValueError()
+    )
+
+    # This should fail since the error is a ValueError
+    with pytest.raises(NotConnected) as e:
+        await dummy_device_two_working_one_timeout_two_value_error.connect(
+            timeout=0.01
+        ),
+    assert e.value._errors == TWO_WORKING_TWO_TIMEOUT_TWO_VALUE_ERROR_OUTPUT
+
+    logs = caplog.get_records("call")
+    logs = [log for log in logs if "ophyd_async" in log.pathname]
+    assert len(logs) == 4
+
+    for i in range(0, 2):
+        assert (
+            "device `value_error_signal` raised unexpected exception ValueError"
+            == logs[i].message
+        )
+        assert logs[i].levelname == "ERROR"
+
+    assert logs[2].levelname == "DEBUG"
+    assert logs[3].levelname == "DEBUG"
+    # These messages could come in any order
+    messages = [logs[idx].message for idx in (2, 3)]
+    for protocol in ("pva", "ca"):
+        assert f"signal {protocol}://A_NON_EXISTENT_SIGNAL timed out" in messages
+
+
+async def test_error_handling_device_collector(caplog):
+    caplog.set_level(10)
+    with pytest.raises(NotConnected) as e:
+        async with DeviceCollector(timeout=0.1):
+            # flake8: noqa
+            dummy_device_two_working_one_timeout_two_value_error = (
+                DummyDeviceTwoWorkingTwoTimeOutTwoValueError()
+            )
+            dummy_device_one_working_one_timeout = DummyDeviceOneWorkingOneTimeout()
+
+    expected_output = {
+        "dummy_device_two_working_one_timeout_two_value_error": TWO_WORKING_TWO_TIMEOUT_TWO_VALUE_ERROR_OUTPUT,
+        "dummy_device_one_working_one_timeout": ONE_WORKING_ONE_TIMEOUT_OUTPUT,
+    }
+    assert expected_output == e.value._errors
+
+    logs = caplog.get_records("call")
+    logs = [log for log in logs if "ophyd_async" in log.pathname]
+    assert len(logs) == 5
+    assert (
+        logs[0].message
+        == logs[1].message
+        == "device `value_error_signal` raised unexpected exception ValueError"
+    )
+    assert logs[0].levelname == logs[1].levelname == "ERROR"
+
+    assert logs[2].levelname == "DEBUG"
+    assert logs[3].levelname == "DEBUG"
+    # These messages could come in any order
+    messages = [logs[idx].message for idx in (2, 3)]
+    for protocol in ("pva", "ca"):
+        assert f"signal {protocol}://A_NON_EXISTENT_SIGNAL timed out" in messages
+
+
+def test_not_connected_error_output():
+    not_connected = NotConnected(TWO_WORKING_TWO_TIMEOUT_TWO_VALUE_ERROR_OUTPUT)
+
+    assert str(not_connected) == (
+        "timeout_child_device_ca:\n"
+        "    timeout_signal: signal ca://A_NON_EXISTENT_SIGNAL timed out\n"
```

Comment:
Please can we include the error type here
```suggestion
        "    timeout_signal: NotConnected: ca://A_NON_EXISTENT_SIGNAL\n"
```